path-finding.asciidoc

```diff
@@ -12,3 +12,31 @@ Relevant questions to answer:
   * What information can be sent to intermediate and the final node aside from the critical routing data?
   * What are multi-hop locks? What addition privacy and security guarantees to they offer?
   * How can the flexible onion space be used to enabled packet switching in the network?
+
+
+=== What is "Source-Based" routing and why does the Lightning Network use it?
+
+Source-based routing is a method of path-finding where the sender (i.e. the source) plans the route from itself, through the intermediary nodes, and finally to the destination.
+Once a route has been selected, the sender sends the payment to the first intermediary node, who sends it to the second intermediary node and so on, until it reaches the destination.
+While a payment is in-flight along a route, the route cannot be changed by any of the intermediary nodes, even if a shorter path or a cheaper path (in terms of routing fees) exists.
+
+The Lightning Network uses source-based routing at the time of writing in order to protect user privacy.
+As discussed in the chapter on Onion Routing, the intermediary nodes transmitting the payment are not aware of the full route of the payment; they only know the node they received it from and the node they are sending it to.
+As such, intermediary nodes cannot assist with path-finding because this requires them to know the destination of the payment.
+Without knowing the destination, they cannot calculate whether the path selected is efficient or not.
+
+We also cannot expect the destination to find a path.
+Like Bitcoin, the Lightning Network uses a "push" system to make payments.
+In Bitcoin, payments are pushed by the sender to the receiver's address, and in Lightning, payments are pushed from the sender's channel to fulfill the receiver's invoice.
```

Comment:
to simplistic and semantically maybe not fitting in the source based routing section? 

In lightning HTLCs are set up from sender to receiver along a path of channels but once the path of htlcs is set up the recipient actually pulls the payment by setteling the htlcs which propagates back by a chain of reimbursments all the way to the sender

Reply:
Yeah you're right, I oversimplified it a little too drastically. Decided to just kill those two lines. BTW looks like I did specify that the destination can provide routing hints below that line so removed the first mention of it.

https://github.com/lnbook/lnbook/pull/309/commits/56a7b75c0d6627f52f4f1bf49329b7c9ab21d5ff

Reply:
I think for here this is much better. the telescoping will be explained elsewhere (hopefully / probably)